lib/modules/EyeglassModules.js

```diff
@@ -257,39 +260,34 @@ function canAccessModule(name, origin) {
   var canAccessFrom = function canAccessFrom(origin) {
     // find the nearest package for the origin
     var pkg = packageUtils.findNearestPackage(origin);
-
-    var cache = this.cache.access;
-    cache[name] = cache[name] || {};
-
-    // check for cached result
-    if (cache[name][pkg] !== undefined) {
-      return cache[name][pkg];
-    }
-
-    // find all the branches that match the origin
-    var branches = findBranchesByPath({
-      dependencies: this.tree
-    }, pkg);
-
-    var canAccess = branches.some(function(branch) {
-      // if the reference is to itself (branch.name)
-      // OR it's an immediate dependency (branch.dependencies[name])
-      if (branch.name === name || branch.dependencies && branch.dependencies[name]) {
-        return true;
-      }
+    var cacheKey = JSON.stringify({
```

Comment:
stringify isn't stable by default, the cache key may not result in collisions due to this.

would `${origin}|${pkg.name}|${pkg.version}` be sufficient, or better yet `realpath` of that location?

If not, maybe [the following lib](https://github.com/substack/json-stable-stringify) may be good.